terraform/ec2/linux/main.tf

```diff
@@ -26,6 +26,51 @@ locals {
   private_key_content = var.ssh_key_name != "" ? var.ssh_key_value : tls_private_key.ssh_key[0].private_key_pem
 }
 
+#####################################################################
+# Create EFS
+#####################################################################
+resource "aws_efs_file_system" "efs" {
+  creation_token = "efs-${random_id.testing_id.hex}"
+  tags           = {
+    Name = "efs-${random_id.testing_id.hex}"
+  }
+}
+
+resource "aws_efs_mount_target" "mount" {
+  file_system_id = aws_efs_file_system.efs.id
+  subnet_id = aws_instance.cwagent.subnet_id
+  security_groups = [aws_security_group.ec2_security_group.id]
+}
+
+resource "null_resource" "mount_efs" {
+  depends_on = [
+    aws_efs_mount_target.mount,
+    aws_instance.cwagent
+  ]
+
+  connection {
+    type = "ssh"
+    user = "ec2-user"
+    private_key = local.private_key_content
+    host = aws_instance.cwagent.public_ip
+  }
+
+  provisioner "file" {
```

Comment:
I wonder should we setup all the resources (including cloning) instead of [letting at null resource integration test](https://github.com/williazz/amazon-cloudwatch-agent-test/blob/main/terraform/ec2/linux/main.tf#L32-L43)? Since it would reduce the necessary to use "file" with scp, want to ask your opinion on the design script though.

Reply:
Not sure what you mean by "setup all the resources" instead of using the null resource.

If this answers your question, I plan to move the efs mounting process from terraform to the AMI building pipelines next week, which should dramatically reduce the over-engineering in this PR

Reply:
Then LGTM.